awx/plugins/inventory/foreman.py

```diff
@@ -203,6 +249,44 @@ def _get_json(self, url, ignore_errors=None, params=None):
                 break
         return results
 
+    def _use_inventory_report(self):
+         url = "%s/api/v2/status" % self.foreman_url
+         result = self._get_json(url)
+         foreman_version = LooseVersion(result.get('version')) >= LooseVersion('1.24.0')
+         return foreman_version and self.foreman_use_reports_api
+
+    def _fetch_params(self):
+        options, params = ("no", "yes") , dict()
+        params["Organization"] = options[self.want_organization]
+        params["Location"] = options[self.want_location]
+        params["IPv4"] = options[self.want_IPv4]
+        params["Facts"] = options[self.want_facts]
+        params["Host Group"] = options[self.want_host_group]
+        params["Host Collections"] = options[self.want_hostcollections]
+        params["Subnet"] = options[self.want_subnet]
+        params["Smart Proxies"] = options[self.want_smart_proxies]
+        params["Content Attributes"] = options[self.want_content_facet_attributes]
+        return params
+
+    def _post_request(self):
+        url = "%s/ansible/api/v2/ansible_inventories/schedule" % self.foreman_url
+        session = self._get_session()
+        params = {'input_values': self._fetch_params()}
+        ret = session.post(url, json=params)
+        url = "{0}/api/v2/report_templates/153-Ansible%20Inventory/report_data/{1}".format(self.foreman_url, ret.json().get('job_id'))
```

Comment:
hardcoding 153 wouldn't work here, you can use data_url from response though, this helped in my case
```
url = "{0}/{1}".format(self.foreman_url, ret.json().get('data_url'))
```